modeling-cmds/src/def_enum.rs

```diff
@@ -1061,6 +1061,43 @@ define_modeling_cmd_enum! {
             Clone, Debug, Deserialize, JsonSchema, Serialize, ModelingCmdVariant,
         )]
         pub struct GetNumObjects;
+
+        ///Set the parent object of an object
+        #[derive(
+            Clone, Debug, Deserialize, JsonSchema, Serialize, ModelingCmdVariantEmpty,
+        )]
+        pub struct SetObjectParent
+        {
+            ///The object (child) whose parent will be set
+            pub object_id: Uuid,
+            /// Id of the new parent, if null will unset any parent of object_id
+            pub parent_id: Option<Uuid>
+        }
+
+        #[derive(Clone, Debug, Deserialize, JsonSchema, Serialize)]
+        struct TransformBy<T> {
+            /// The scale, or rotation, or translation.
+            property: T,
+            /// If true, overwrite the previous value with this.
+            /// If false, the object will be scaled, or translated, or rotated, etc.
+            set: bool,
+        }
+
+        ///Set the transform of an object.
+        #[derive(
+            Clone, Debug, Deserialize, JsonSchema, Serialize, ModelingCmdVariantEmpty,
+        )]
+        pub struct SetObjectTransform
+        {
+            ///Id of the object whose transform is to be set
+            pub object_id: Uuid,
+            #[serde(default)]
+            translate: Option<TransformBy<Point3d>>,
+            #[serde(default)]
+            rotate: Option<TransformBy<Point3d>>,
+            #[serde(default)]
+            scale: Option<TransformBy<Point3d>>,
+        }
```

Comment:
The following is just a suggestion. We can further reduce bandwidth by reducing the structure size and instead sending a list of transforms. Otherwise looks great!

```suggestion
enum TransformOperation {
    Translate,
    Rotate,
    Scale
}

struct Transform<T> {
    operation: TransformOperation,
    value: T,
    set: bool,
}

///Set the transform of an object.
#[derive(
    Clone, Debug, Deserialize, JsonSchema, Serialize, ModelingCmdVariantEmpty,
)]
pub struct SetObjectTransform
{
    ///Id of the object whose transform is to be set
   pub object_id: Uuid,
   transform: Vec<Transform<Point3D>>
}
```

Reply:
I thought about this, but this allows you to send multiple transforms that overwrite each other, e.g. [(Translate, <0,0,0>), (Translate, <1,1,1>)]. So I think the current approach is better -- it limits you to sending at most one transform for a given property and the given project.

Reply:
It could be desirable from a programmer or mathematician point of view to compose transforms in various ways, ways I cannot imagine - for example `[(Translate, <0,0,0>), (Translate, <1,1,1>)].` could be desirable if they are composing and have `set: false` . Otherwise I agree with you.

Reply:
You are knocking on the door of a transform stack here. Which is where you specify a chain of transforms one after the other that get applied in that order. Typically comes along with the ability to push and pop transforms to a stack that lets you get pretty flexible

Reply:
We're now agreed that Transform endpoint should take a list of transforms.